cli/start.go

```diff
@@ -140,7 +140,7 @@ func MakeStartCommand() *cobra.Command {
 				// TODO-ACP: Infuture when we add support for the --no-acp flag when admin signatures are in,
 				// we can allow starting of db without acp. Currently that can only be done programmatically.
 				// https://github.com/sourcenetwork/defradb/issues/2271
-				dbOpts = append(dbOpts, db.WithACP(rootDir))
+				acpOpts = append(acpOpts, node.WithACPPath(rootDir))
```

Comment:
question/nitpick: Why not maintain the InMem and non InMem variants? if getting rid because it seems specific to local acp, then might as well also change `node.WithACPPath` to  `node.WithACPTarget`?

Reply:
> Why not maintain the InMem and non InMem variants?

It just didn't seem to be worth keeping atm (increased surface area, and we would never reference it ourselves).  Long-term it's current implementation doesn't really make that much sense either as it doesn't have to be local-acp specific either.  Better to have a proper think about it if/when the demand for it ever arises.

Reply:
I guess that makes sense, unless a user wanted to use specifically in-mem or persistence acp model that didn't match the type of db persistence we want.

For now we can just assume, for an in-mem DB, we use in mem ACP, and vice-versa